src/main/java/gregtech/common/covers/filter/FilterTypeRegistry.java

```diff
@@ -42,6 +42,17 @@ public static void registerItemFilter(int id, Class<? extends ItemFilter> itemFi
         itemFilterById.put(id, itemFilterClass);
     }
 
+    public static int getIdForFilter(Object filter) {
```

Comment:
Why do we even have this class? Is this actually needed for anything with this rework? I don't think filters need a registry, they should be able to function without being explicitly registered.

Reply:
there needs to be a way to retrieve/create a filter for a given filter item stack. it might be possible to add the filter to the filter item in `MetaItem` as a capability or something